projects/dsp-ui/src/lib/viewer/representation/still-image/still-image.component.scss

```diff
@@ -0,0 +1,85 @@
+// sizes
+$max-width: 800px;
+$nav-width: 36px;
+
+$osd-width: calc(100% - (2 * #{$nav-width}));
+$osd-max-width: calc(#{$max-width} - (2 * #{$nav-width}));
+
+$osd-height: 400px;
+$full-height: calc(#{$osd-height} + 64px + 24px);
+
+// colors
+$dark: #000;
+$bright: #ccc;
+
+.still-image-viewer {
+  display: inline-flex;
+  height: $osd-height;
+  max-width: $max-width;
+  width: 100%;
+
+  @media (max-height: 636px) {
+    height: 300px;
+  }
+
+  .navigation {
+    height: $full-height;
+    width: $nav-width;
+
+    .mat-button.full-size {
+      height: 100% !important;
+      width: $nav-width !important;
+
+      padding: 0 !important;
+      min-width: $nav-width !important;
+    }
+  }
+
+  .content {
+    height: $full-height;
+    max-width: $osd-max-width;
+    width: $osd-width;
+
+    .osd-container {
+      color: $bright;
+      background-color: $dark;
+      height: $osd-height;
+
+      &.fullscreen {
+        max-width: 100vw;
+      }
+    }
+
+    .footer p {
+      color: $bright;
+      background-color: $dark;
+      height: 24px;
+      margin: 0;
+      padding: 0 16px;
+    }
+  }
+}
+
+
+/*
+ Overlay styling
+ */
+
+::ng-deep .roi-svgoverlay {
```

Comment:
is there a way to accomplish this without using ::ng-deep? We should avoid using it since it will be/has been(?) deprecated. No one seems to know what's going with ::ng-deep https://github.com/angular/angular/issues/25160

Reply:
To be honest I don't know. I assume that there was a CSS problem with the SVG overlays.

The docs say:

> Component styles normally apply only to the HTML in the component's own template.
Applying the ::ng-deep pseudo-class to any CSS rule completely disables view-encapsulation for that rule. Any style with ::ng-deep applied becomes a global style. In order to scope the specified style to the current component and all its descendants, be sure to include the :host selector before ::ng-deep. If the ::ng-deep combinator is used without the :host pseudo-class selector, the style can bleed into other components.

https://angular.io/guide/component-styles#view-encapsulation

The thing is that OpenSeadragon is not an Angular component so I wonder wether `::ng-deep` is necessary. I will try to remove it and see what happens.

Reply:
`::ng-deep` is deprecated but there does not seem to be a good alternative for it, see https://github.com/angular/angular/issues/25160

However, on way to avoid `::ng-deep` is modifying the components view encapsulation, see
- https://github.com/angular/angular/issues/25160#issuecomment-504642697
- https://stackoverflow.com/questions/47024236/what-to-use-in-place-of-ng-deep

By using the component's selector in the CSS rules, these should only take effect for the component. This is important because changing view encapsulation makes the rules **global** which contradicts the concept of the component encapsulation.

Reply:
I created https://dasch.myjetbrains.com/youtrack/issue/DSP-517 which documents this alternative to `::ng-deep`. We decided to use `::ng-deep` for now because disabling view encapsulation seems conceptually wrong to us.